addons/main/functions/fnc_getLauncherUnits.sqf

```diff
@@ -0,0 +1,73 @@
+#include "script_component.hpp"
+/*
+ * Author: ThomasAngel
+ * Gets all units from a group that have launchers.
+ * Things like AP launchers, flares etc will not count, -
+ * launcher must be capable of destroying at least some vehicle.
+ *
+ * Checking through the submunitions might improve mod compatibility,
+ * but only if the AI actually accounts for submunitions when evaluating -
+ * whether to engage or not. As of writing, this is unknown.
+ *
+ * Disposable launchers get mostly ignored, AI doesn't seem to even use them.
+ *
+ * Arguments:
+ * 0: Group to search in <GROUP>
+ * 1: Check through submunitions <BOOLEAN>
+ *
+ * Return Value:
+ * Array - units with launchers
+ *
+ * Example:
+ * [group bob] call lambs_main_fnc_getLauncherUnits;
+ *
+ * Public: Yes
+*/
+
+#define LIGHT_VEHICLE "128"
+#define AIR_VEHICLE "256"
+#define HEAVY_VEHICLE "512"
+
+params [
+    ["_group", grpNull, [grpNull]],
+    ["_checkSubmunition", false, [false]],
+    ["_offensiveVeh", true, [true]],
+    ["_offensiveAir", true, [true]],
+    ["_offensiveArmor", true, [true]]
+];
+
```

Comment:
```suggestion
private _cfgMagazines = configFile >> "cfgMagazines";
private _cfgAmmo = configFile >> "cfgAmmo";

```

Reply:
Did you benchmark this? I benchmarked this, but the config lookups were so fast that making the paths into variables actually caused a slowdown. Something like ~0.220ms -> ~0.225ms. Tested on a vanilla NATO weapons squad, both without any changes and with the only launcher-equipped unit deleted, in both cases the version without the config path variables was faster.

A niche situation where the path variables might be faster is if every single squadmember has a secondary weapon that isn't suitable, but I think that's a very unconventional situation as squads by default rarely have more than 1 to 2 soldiers with a launcher, even in common faction mods.

Reply:
> Did you benchmark this?

Nope.

> making the paths into variables actually caused a slowdown.

Huh, interesting. Good to know.